src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -607,32 +612,27 @@ static int guigetstr(Object* ho, char** cpp) {
     return 1;
 }
 
-static PyObject* loads;
-static PyObject* dumps;
+static nb::callable loads;
+static nb::callable dumps;
 
 static void setpickle() {
     if (dumps) {
         return;
     }
-    PyObject* pickle = PyImport_ImportModule("pickle");
-    if (pickle) {
-        Py_INCREF(pickle);
-        dumps = PyObject_GetAttrString(pickle, "dumps");
-        loads = PyObject_GetAttrString(pickle, "loads");
-        if (dumps) {
-            Py_INCREF(dumps);
-            Py_INCREF(loads);
-        }
-    }
+    nb::module_ pickle = nb::module_::import_("pickle");
+    dumps = pickle.attr("dumps");
+    loads = pickle.attr("loads");
     if (!dumps || !loads) {
         hoc_execerror("Neither Python cPickle nor pickle are available", 0);
     }
+    dumps.inc_ref();
```

Comment:
```suggestion
    // We intentionally leak these, because if we don't
    // we observe SEGFAULTs during application shutdown.
    // Likely because `~nb::callable` runs after the Python
    // library cleaned up.
    dumps.inc_ref();
```